lib/merchant_repository.rb

```diff
@@ -1,41 +1,26 @@
-class MerchantRepository
-  # This is the MerchantRepository Class
-  # frozen_string_literal: true.
-  attr_reader :merchants
+# This is the MerchantRepository Class
+# frozen_string_literal: true.
+require_relative 'merchant'
+require_relative 'general_repo'
 
-  def initialize
-    @merchants = []
-  end
+class MerchantRepository < GeneralRepo
+  attr_reader :repository
 
-  def create(data)
-    data[:id] ||= 1 if merchants == []
-    data[:id] ||= (@merchants.last.id.to_i + 1).to_s
-    @merchants << Merchant.new(data)
+  def initialize(data = {})
+    super(data)
   end
 
-  def all
-    @merchants
-  end
-
-  def find_by_id(id)
-    merchants.select { |merchant| merchant.id == id }.first
+  def create(data)
+    data[:id] ||= 1 if repository == []
+    data[:id] ||= (@repository.last.id.to_i + 1).to_s
+    @repository << Merchant.new(data)
   end
 
   def find_by_name(name)
-    merchants.select { |merchant| merchant.name.casecmp?(name) }.first
+    repository.find { |merchant| merchant.name.casecmp?(name) }
   end
 
   def find_all_by_name(name)
-    merchants.select { |merchant| merchant.name.downcase.include? name.downcase }
-  end
-
-  def update(id, name)
-    merchant = find_by_id(id)
-    merchant.update(name)
-  end
-
-  def delete(id)
-    to_be_deleted = find_by_id(id)
-    @merchants.delete(to_be_deleted)
+    repository.select { |merchant| merchant.name.downcase.include? name.downcase }
```

Comment:
I am still not convinced this is the behavior we want. If I am reading this correctly the condition on which we select is asking if the argument we passed in is contained within merchant.name. I understood the behavior we were after is more along the lines of 'merchant.name.downcase == name.downcase'

Reply:
This is written like this because of the wording in iteration 0: "find_all_by_name(name) - returns either [] or one or more matches which **contain** the supplied **name fragment**, case insensitive". To me this wording implies that it can be fed a partial name and return a list of all that contain that partial name.